pkg/tsdb/azuremonitor/credentials.go

```diff
@@ -42,44 +42,37 @@ func getAuthType(cfg *setting.Cfg, jsonData *simplejson.Json) string {
 	}
 }
 
-func getDefaultAzureCloud(cfg *setting.Cfg) (string, error) {
+func getDefaultAzureCloud(cfg *setting.Cfg) string {
 	// Allow only known cloud names
-	cloudName := cfg.Azure.Cloud
+	cloudName := ""
+	if cfg != nil && cfg.Azure != nil {
+		cloudName = cfg.Azure.Cloud
+	}
 	switch cloudName {
 	case azsettings.AzurePublic:
-		return azsettings.AzurePublic, nil
+		return azsettings.AzurePublic
 	case azsettings.AzureChina:
-		return azsettings.AzureChina, nil
+		return azsettings.AzureChina
 	case azsettings.AzureUSGovernment:
-		return azsettings.AzureUSGovernment, nil
-	case azsettings.AzureGermany:
-		return azsettings.AzureGermany, nil
-	case azsettings.AzureCustomized:
-		return azsettings.AzureCustomized, nil
+		return azsettings.AzureUSGovernment
 	case "":
 		// Not set cloud defaults to public
-		return azsettings.AzurePublic, nil
+		return azsettings.AzurePublic
 	default:
-		err := fmt.Errorf("the cloud '%s' not supported", cloudName)
-		return "", err
+		return cloudName
 	}
 }
 
-func normalizeAzureCloud(cloudName string) (string, error) {
+func normalizeAzureCloud(cloudName string) string {
```

Comment:
Now the logic is almost the same as that of `azsettings.NormalizeAzureCloud`. If no check is applied, can use `azsettings.NormalizeAzureCloud` instead.

Reply:
`azsettings.NormalizeAzureCloud` uses different keys for the clouds. For example, `AzurePublic` here is "azuremonitor", which is not an option in the function that the SDK exposes :disappointed: